tools/cli/helpers/dependencyAnalysis.js

```diff
@@ -186,3 +186,44 @@ export function getBuildOrder( deps ) {
 
 	return ret;
 }
+
+/**
+ * Get the max "depth" of all projects from a set of targets.
+ *
+ * If there are cycles, the "depth" for affected projects will be `deps.size + 1`.
+ * Any projects not reachable from any target will have a depth of Infinity.
+ *
+ * @param {Map}      deps    - Project dependency map.
+ * @param {Iterable} targets - Targets.
+ * @return {Map<string,number>} - Map of projects to depths.
+ */
+export function getDependencyDepths( deps, targets ) {
+	const depths = new Map();
+
+	for ( const p of deps.keys() ) {
+		depths.set( p, Infinity );
+	}
+
+	const maxDepth = deps.size + 1;
```

Comment:
If we hit this, there are probably other issues. 😃

Reply:
Yeah. It's here just to keep it from looping forever.